annotated_types/__init__.py

```diff
@@ -170,15 +171,24 @@ class GroupedMetadata:
     - `Annotated[int, *Field(...)]` (PEP-646)
     """  # noqa: trailing-whitespace
 
-    __slots__ = ()
-
-    def __init_subclass__(cls, *args: Any, **kwargs: Any) -> None:
-        super().__init_subclass__(*args, **kwargs)
-        if cls.__iter__ is GroupedMetadata.__iter__:
-            raise TypeError("Can't subclass GroupedMetadata without implementing __iter__")
+    @property
+    def __is_annotated_types_grouped_metadata__(self) -> Literal[True]:
+        return True
```

Comment:
Note that this can be implemented as an attribute, the usual "cannot override a property with an attribute" thing type checkers like to complain about doesn't apply to protocol subclasses.